src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -337,6 +338,54 @@ def data_arguments(self):
             "Abstract property. Which symbols are data arguments is"
             " API-specific.")
 
+    def copy_external_global(self, globalvar):
+        '''
+        Copy the given global variable (and its referenced ContainerSymbol if
+        needed) into the SymbolTable.
+
+        :param globalvar: the variable to be copied in
+        :type globalvar: :py:class:`psyclone.psyir.symbols.DataSymbol`
+
+        :raise TypeError: if the given variable is not a global variable
+        :raise KeyError: if the given variable name already exists in the \
+            symbol table
+        '''
+
+        if not isinstance(globalvar, DataSymbol):
+            raise TypeError(
+                "The globalvar argument of SymbolTable.copy_external_global"
+                " method should be a DataSymbol, but found {0}."
+                "".format(type(globalvar)))
+
+        if not globalvar.is_global:
+            raise TypeError(
+                "The globalvar argument of SymbolTable.copy_external_"
+                "global method should have a GlobalInterface interface, "
+                "but found {0}.".format(type(globalvar.interface)))
```

Comment:
Since the rest of the code mostly uses `type(xxx).__name__` I will use this in all the introduced error messages, but I am not sure if I agree that it's nicer :)